### PR TITLE
Explicit exit-phase coverage for StageConditionGate (#315)

### LINKED ISSUE #315: Verify stage-level `conditions:` reliably skip stages, including edge cases
## Summary

Both `gameStage` and `introExit` step schemas accept a `conditions:` field at the stage level (see [`treatment.ts`](packages/stagebook/src/schemas/treatment.ts)). The intent — based on the schema and the way it's used in real treatments — is that when the conditions evaluate false, the entire stage is skipped: the participant advances to the next stage in the sequence without rendering, blocking, or producing phantom data records for the skipped stage.

This is a load-bearing assumption for any treatment that uses an optional path through the experiment (e.g. an opt-in second discussion gated on a binary preference). The schema accepts it and a treatment can be authored against it, but the runtime behavior — particularly around edge cases — doesn't appear to have explicit test coverage. Filing this to (a) surface the question, (b) propose concrete test cases that would resolve it, and (c) ensure the behavior is documented for treatment authors who depend on it.

## Motivating use case

\`dialogue-levers/pilot_3\` includes an optional 15-minute \`second_discussion\` gameStage that fires only when both members of a dyad answered \"Yes\" to a \`continue_with_partner\` binary in a preceding stage. The stage-level condition is:

\`\`\`yaml
- name: second_discussion
  conditions:
    - reference: 0.prompt.continue_with_partner
      comparator: equals
      value: \"Yes\"
  duration: 900
  discussion:
    chatType: video
    showNickname: true
    showTitle: false
  elements:
    - type: prompt
      file: game/second_discussion_instructions.prompt.md
    # ...
\`\`\`

For dyads where the condition is false, this stage should be skipped end-to-end — no video room created, no rendering, no waiting indicator. The participants should advance straight to the next stage.

A second similar case fires in the same treatment's \`exitSequence\`: a \`second_discussion_followup\` survey gated on the same condition, so participants who didn't have a second discussion don't see the followup items.

### 1. Clean-skip behavior

A 3-stage treatment where stage 2 has a false stage-level condition. Expected: participant goes from stage 1 → stage 3, skipping stage 2 entirely. Specifically:

- No render of stage 2's elements (no DOM mount, no API calls for stage 2's prompts).
- No \"waiting for your partner\" indicator while the other side resolves the same skip.
- No record of stage 2 in the participant's flow data (or, if a record is written, it's clearly marked as \`skipped: true\` or equivalent so downstream analysis can distinguish \"skipped\" from \"missing\").
- Stage 3 receives all the data context it needs (prompts from stage 1 remain referenceable; nothing assumes stage 2 ran).
- Time accounting: the skipped stage doesn't add to elapsed-time reporting.

### 2. Partner-attrited condition evaluation

In a \`playerCount: 2\` treatment, the condition \`0.prompt.continue_with_partner == \"Yes\"\` references a value that depends on position 0 having answered. What happens if position 0 attrits before answering, but position 1 reaches the gating stage?

Possibilities:

- **(a) Condition resolves false → stage skipped.** This is the desired behavior — the remaining participant proceeds normally.
- **(b) Condition resolves to undefined / null → ?** Implementation-dependent; could throw, hang, render the stage anyway, or silently skip. We need to know which.
- **(c) Stage rendering errors out, participant is trapped.** Worst case; would need a host-level recovery mechanism.

The fix (if not already (a)) is probably to define \`reference-resolves-to-undefined\` as condition-falsy in the comparator semantics. That covers attrition, partial completion, and any other \"the data isn't there yet\" case under one rule.

### 3. Stage-level condition with boolean-tree operators

After [#235] (boolean tree in conditions), stage-level conditions can use \`all:\` / \`any:\` / \`none:\`. Worth confirming the same skip behavior holds when the gating predicate is \`all: [...]\` over multiple references, with various combinations of true/false/undefined inputs.

### 4. Ordering: condition reference to a stage that itself was skipped

Pathological but possible: stage A is conditionally skipped; stage B's gating condition references a prompt that would have been written by stage A. What does the condition resolve to? Should be falsy by the same rule as #2.

## Suggested tests (drop into existing test suite)

These can be added to whichever test file exercises stage-level condition semantics — possibly a new \`stageConditions.test.ts\` or folded into the existing condition / treatment tests.

\`\`\`ts
import { expect, test } from \"vitest\";
// adjust imports to whatever the runtime helper is for \"advance the
// participant through a treatment and observe which stages render\"

test(\"stage with false stage-level condition is skipped end-to-end\", () => {
  const treatment = {
    treatments: [{
      name: \"skip_demo\",
      playerCount: 1,
      gameStages: [
        {
          name: \"set_value\",
          duration: 10,
          elements: [
            { type: \"prompt\", file: \"value.prompt.md\", name: \"gate\" },
            { type: \"submitButton\", buttonText: \"Continue\" },
          ],
        },
        {
          name: \"should_skip\",
          conditions: [
            { reference: \"self.prompt.gate\", comparator: \"equals\", value: \"skip_me\" },
          ],
          duration: 10,
          elements: [
            { type: \"prompt\", file: \"skipped.prompt.md\", name: \"skipped_field\" },
            { type: \"submitButton\", buttonText: \"Continue\" },
          ],
        },
        {
          name: \"after_skip\",
          duration: 10,
          elements: [
            { type: \"prompt\", file: \"after.prompt.md\", name: \"after_field\" },
            { type: \"submitButton\", buttonText: \"Continue\" },
          ],
        },
      ],
    }],
  };

  const flow = simulateParticipantFlow(treatment, {
    answers: { gate: \"do_not_skip\" },
  });

  expect(flow.stagesRendered).toEqual([\"set_value\", \"should_skip\", \"after_skip\"]);
});

test(\"stage with true skip condition does not render and advances cleanly\", () => {
  const treatment = /* same as above */;

  const flow = simulateParticipantFlow(treatment, {
    answers: { gate: \"skip_me\" },
  });

  expect(flow.stagesRendered).toEqual([\"set_value\", \"after_skip\"]);
  expect(flow.stagesSkipped).toEqual([\"should_skip\"]);
  // no API calls, no DOM mount, no time accumulated for \"should_skip\"
});

test(\"stage condition referencing an unanswered prompt resolves to falsy (skip)\", () => {
  // Models the partner-attrited case: the reference target was never
  // written. Condition should evaluate to false → stage skipped.
  const treatment = {
    treatments: [{
      name: \"attrit_demo\",
      playerCount: 2,
      gameStages: [
        {
          name: \"answer_or_attrit\",
          duration: 10,
          elements: [
            { type: \"prompt\", file: \"ask.prompt.md\", name: \"answer\" },
            { type: \"submitButton\", buttonText: \"Continue\" },
          ],
        },
        {
          name: \"gated_on_partner\",
          conditions: [
            { reference: \"0.prompt.answer\", comparator: \"equals\", value: \"Yes\" },
          ],
          duration: 10,
          elements: [
            { type: \"prompt\", file: \"after.prompt.md\" },
            { type: \"submitButton\", buttonText: \"Continue\" },
          ],
        },
        {
          name: \"always_runs\",
          duration: 10,
          elements: [
            { type: \"prompt\", file: \"final.prompt.md\" },
            { type: \"submitButton\", buttonText: \"Continue\" },
          ],
        },
      ],
    }],
  };

  // Position 0 never answers; position 1 reaches the gated stage.
  const flow = simulateParticipantFlow(treatment, {
    position: 1,
    partnerState: { answered: false },
  });

  expect(flow.stagesRendered).toContain(\"always_runs\");
  expect(flow.stagesRendered).not.toContain(\"gated_on_partner\");
  // crucial: position 1 doesn't get trapped waiting for position 0's
  // answer or seeing an error
});

test(\"exitSequence step with stage-level condition follows the same skip rule\", () => {
  // Same skeleton, but the conditional step is in exitSequence rather
  // than gameStages, since the runtime semantics may differ between
  // sync gameStages and async exit steps.
  const treatment = {
    treatments: [{
      name: \"exit_skip_demo\",
      playerCount: 1,
      gameStages: [/* one stage that sets the gate */],
      exitSequence: [
        {
          name: \"conditional_followup\",
          conditions: [
            { reference: \"self.prompt.gate\", comparator: \"equals\", value: \"show\" },
          ],
          elements: [/* ... */],
        },
        {
          name: \"always_runs_in_exit\",
          elements: [/* ... */],
        },
      ],
    }],
  };

  // Confirm same behavior as gameStage skip in both true and false branches.
});
\`\`\`

Apologies if these duplicate existing coverage — happy to delete this if you can point me at where the behavior is already exercised. If it isn't, the four scenarios above (clean skip, attrited evaluation, boolean-tree gating, transitive skip) cover the cases the dialogue-levers treatment relies on.

## Documentation request

Whatever the resolved behavior is, worth a paragraph in \`docs/researcher/conditions.md\` (or wherever stage-level conditions are documented) covering:

- That \`conditions:\` at the stage level skips the stage when false.
- What \"false\" means for a condition that references unanswered or partial data (resolves to falsy → skip; or whatever the actual rule is).
- Whether the same semantics apply in \`exitSequence\` steps.
- How (or whether) skipped stages appear in the per-participant flow data, so downstream analysis can distinguish skipped from missing.

## Real-world example

\`dialogue-levers/pilot_3.treatments.yaml\` has two stages depending on this behavior (one in gameStages, one in exitSequence). If anything in this issue reveals a divergence between the schema's apparent intent and the runtime, that treatment will be the first to break in production.

### PR DESCRIPTION
**Pull request by JamesPHoughton**
Closes #315.

## Summary

The skip-at-load mechanism was already covered for game stages and (by inference from \`Stage\`'s phase-agnostic design + the viewer's \`flattenSteps\`) for intro and exit phases. This adds an explicit CT test that mirrors the dialogue-levers \`second_discussion_followup\` pattern from the issue: an exit-sequence stage gated on a prior game-phase prompt, asserting the gate skips end-to-end when the referenced data is absent.

That single test simultaneously demonstrates **scenario 1 (clean skip)** and **scenario 4 (transitive skip)** from #315 in an exit-phase context — when the gating stage was itself skipped, its outputs are never written, which is structurally identical to \"data is absent\".

## What was already verified

| # | Scenario | Already covered |
| - | -------- | --------------- |
| 1 | Clean skip when condition is false | [StageConditionGate.ct.tsx:39](packages/stagebook/src/components/conditions/StageConditionGate.ct.tsx#L39) |
| 2 | Undefined reference → falsy | [StageConditionGate.ct.tsx:71](packages/stagebook/src/components/conditions/StageConditionGate.ct.tsx#L71), [evaluateConditions.test.ts:226](packages/stagebook/src/utils/evaluateConditions.test.ts#L226-L243) |
| 3 | Boolean tree (\`all\`/\`any\`/\`none\`) gating | [evaluateConditions.test.ts:390-691](packages/stagebook/src/utils/evaluateConditions.test.ts#L390-L691) |
| 4 | Transitive skip | Covered by the absent-data path (same primitive) |

## What this PR adds

One CT test (\`exit-phase stage with a false condition skips cleanly\`) that:
- Uses a stage config without \`duration:\` — matches how \`flattenSteps\` shapes exit steps.
- References \`0.prompt.continue_with_partner\` — a game-phase value pattern, mirroring the real dialogue-levers fixture.
- Passes no \`stateValues\` — simulating either partner attrition OR the prior gating stage having been itself skipped.
- Asserts the gate is visible with \`data-state=\"advancing\"\` and \`stageContent\` is not rendered.

## Test plan

- [x] \`npm run lint -w stagebook\` — clean
- [x] \`npm run test:ct -w stagebook -- --grep \"StageConditionGate\"\` — all 6 tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/stagebook/src/components/conditions/StageConditionGate.ct.tsx [hunk 1]
```diff
@@ -125,6 +125,49 @@ test.describe("StageConditionGate (#183)", () => {
...
+        // which is what the runtime sees when the prior gating stage
+        // was itself skipped (transitive skip).
+      />,
+    );
+    const gate = page.locator('[data-testid="stage-condition-gate"]');
+    await expect(gate).toBeVisible();
+    await expect(gate).toHaveAttribute("data-state", "advancing");
+    await expect(
+      page.locator('[data-testid="stageContent"]'),
+    ).not.toBeVisible();
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply packages/stagebook/src/components/conditions/StageConditionGate.ct.tsx]: Fixed in e03bdb1 — switched to `not.toBeAttached()`. You're right that the stronger assertion catches the CSS-hidden-mounted regression that `not.toBeVisible()` would silently allow.

Side note: the two pre-existing skip tests in this same file (lines 67 and 96) use the weaker `not.toBeVisible()` for the same assertion. Happy to upgrade those in a follow-up if you'd like uniform coverage.
